packages/ai/core/tool/mcp/mcp-transport.ts

```diff
@@ -1,9 +1,14 @@
-import { MCPTransport, TransportConfig } from './types';
-import { StdioClientTransport } from './mcp-stdio-transport';
+import { MCPClientError } from '../../../errors';
 import { SSEClientTransport } from './mcp-sse-transport';
+import { MCPTransport, TransportConfig } from './types';
 
 export function createMcpTransport(config: TransportConfig): MCPTransport {
-  return config.type === 'stdio'
-    ? new StdioClientTransport(config)
-    : new SSEClientTransport(config);
+  if (config.type === 'stdio') {
+    throw new MCPClientError({
+      message:
+        'The stdio transport configuration has been deprecated in favor of passing in a custom transport.',
+    });
+  }
+
+  return new SSEClientTransport(config);
 }
```

Comment:
move the `MCPTransport` interface here? (personally I find `types` files confusing and prefer to have types, functions, etc in files with matching names)

Reply:
and potentially transportconfig as well

Reply:
related: this also applies for things like `utils.ts` etc. - not a huge fan of such collection files